src/main/java/com/main/service/impl/PersonServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.main.service.impl;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.main.dto.ResponseDto;
+import com.main.entity.Message;
+import com.main.entity.Person;
+import com.main.repository.PersonRepository;
+import com.main.service.PersonService;
+
+@Service
+public class PersonServiceImpl implements PersonService {
+
+	private PersonRepository personRepository;
+
+	public PersonServiceImpl(PersonRepository personRepository) {
+		this.personRepository = personRepository;
+	}
+
+	public ResponseDto<String> findAllMessagesByPersonId(Long personId, String word) {
```

Comment:
Why does the main path ends up in the error and the branching from the condition has the expected results?